src/main/java/au/org/emii/ncdfgenerator/Main.java

```diff
@@ -0,0 +1,93 @@
+package au.org.emii.ncdfgenerator;
+
+import org.apache.commons.cli.*;
+import java.io.*;
+import java.sql.*;
+
+public class Main {
+    public static void usage(Options options) {
+        HelpFormatter formatter = new HelpFormatter();
+        formatter.printHelp("ncdfgenerator", options);
+
+        System.exit(3);
+    }
+
+    public static void main(String[] args) {
+        Options options = new Options();
+
+        options.addOption("u", "username", true, "Database user.");
+        options.addOption("p", "password", true, "Database password.");
+        options.addOption("d", "db", true, "Database connection string.");
+        options.addOption("D", "driver", true, "Database driver class.");
+        options.addOption("c", "cql", true, "CQL filter to apply.");
+        options.addOption("P", "profile", true, "Profile to use.");
+        options.addOption("o", "output", true, "Output file to use (output.zip as default).");
+        options.addOption("t", "tmp-dir", true, "Set temporary directory for operation.");
+
+        CommandLineParser parser = new DefaultParser();
+        CommandLine cmd = null;
+        try {
+            cmd = parser.parse(options, args);
+        }
+        catch (ParseException e) {
+            usage(options);
+        }
+
+        String username = cmd.getOptionValue("u");
+        String password = cmd.getOptionValue("p");
+        String connectionString = cmd.getOptionValue("d");
+        String databaseDriver = cmd.getOptionValue("D", "org.postgresql.Driver");
+        String cql = cmd.getOptionValue("c");
+        String profile = cmd.getOptionValue("P");
+        String outputFile = cmd.getOptionValue("o", "output.zip");
+        String tmpDir = cmd.getOptionValue("t", System.getProperty("java.io.tmpdir"));
+
+        if (null == username) { usage(options); }
+        if (null == password) { usage(options); }
+        if (null == connectionString) { usage(options); }
+        if (null == databaseDriver) { usage(options); }
+        if (null == cql) { usage(options); }
+        if (null == profile) { usage(options); }
```

Comment:
Not sure if it would be cleaner to aggregate these into one expression eg. if( null == username || null == username ... ) { ... } . ?

I also don't like the reverse ordering of null equality test, although I understand the reasoning - to fail if assignment is used instead. I think it sacrifices readability.